storage/oasis/nodeapi/universal_runtime.go

```diff
@@ -135,8 +136,22 @@ func (rc *UniversalRuntimeApiLite) GetEventsRaw(ctx context.Context, round uint6
 	return evs, nil
 }
 
-func (rc *UniversalRuntimeApiLite) EVMSimulateCall(ctx context.Context, round uint64, gasPrice []byte, gasLimit uint64, caller []byte, address []byte, value []byte, data []byte) ([]byte, error) {
-	return rc.sdkClient.Evm.SimulateCall(ctx, round, gasPrice, gasLimit, caller, address, value, data)
+func (rc *UniversalRuntimeApiLite) EVMSimulateCall(ctx context.Context, round uint64, gasPrice []byte, gasLimit uint64, caller []byte, address []byte, value []byte, data []byte) (*FallibleResponse[[]byte], error) {
+	res, err := rc.sdkClient.Evm.SimulateCall(ctx, round, gasPrice, gasLimit, caller, address, value, data)
+	if errors.Is(err, ErrSdkEVMExecutionFailed) || errors.Is(err, ErrSdkEVMReverted) {
+		localErr := sdkToLocalErr[err]
```

Comment:
What if the key lookup fails? (It especially might fail because of the whole `==` vs `Is`)

I suggest we simplify this: We don't particularly need the "local" error types because nobody is looking at these errors downstream (and we're not writing an SDK where we'd need to worry about 3rd party devs). We can just assign `DeterministicErr: &DeterministicErr{ msg: err.Error() }` and get rid of the local types and the map.

Reply:
oo good idea - updated